src/ssl.c

```diff
@@ -48370,6 +48370,357 @@ int wolfSSL_PEM_write_bio_PKCS7(WOLFSSL_BIO* bio, PKCS7* p7)
     }
     return WOLFSSL_FAILURE;
 }
+
+/******************************************************************************
+* wolfSSL_SMIME_read_PKCS7 - Reads the given S/MIME message and parses it into a PKCS7 object.
+* In case of a multipart message, stores the signed data in bcont.
+*
+* RETURNS:
+* returns pointer to a PKCS7 structure on success, otherwise returns NULL
+*/
+WOLFSSL_API PKCS7* wolfSSL_SMIME_read_PKCS7(WOLFSSL_BIO* in, WOLFSSL_BIO** bcont)
```

Comment:
Please sanitize input arguments for NULL, return BAD_FUNC_ARG if so.  Looks like "in" gets checked later on by wolfSSL_BIO_get_len(), but "bcont" doesn't before getting used.

Reply:
bcont is a special case in that it's a pointer this function fully controls and returns, the definition passed in is not used.
According to the openSSL documentation on this function, this should actually accept bcont as NULL in some cases (see https://www.openssl.org/docs/man1.0.2/man3/SMIME_read_PKCS7.html).  I'm not sure if I can perform any sanitization on it.